frontend/src/pages/CoursesList/CourseActions/CourseActions.tsx

```diff
@@ -32,35 +32,46 @@ const CourseActions: React.FC<ICourseActions> = ({
   targetLoading,
   targetId,
   handleApplyCourse,
-}) => (
-  <CourseActionsBox key={`${course._id}_box`}>
-    <CourseActionsWrapper key={`${course._id}_actions`}>
-      <Link to={chooseListPath(course, index, clientCourses)}>
-        <CustomButton color="primary" variant="mediumOutlined">
+}) => {
+  const isButtonContentLoading = targetLoading && targetId === course._id;
+  return (
+    <CourseActionsBox key={`${course._id}_box`}>
+      <CourseActionsWrapper key={`${course._id}_actions`}>
+        <CustomButton
+          color="primary"
+          variant="mediumOutlined"
+          component={Link}
+          to={chooseListPath(course, index, clientCourses)}
+        >
           {ButtonLabels.details}
         </CustomButton>
-      </Link>
-      {!isAdmin &&
-        (targetLoading && targetId === course._id ? (
-          <CustomButton variant="mediumOutlined" disabled>
-            <ButtonLoader buttonSpinner={buttonSpinner} />
-          </CustomButton>
-        ) : clientCourses ? (
-          <ActionButton
-            label={COURSE_LABELS[clientCourses[index].status]}
-            status={clientCourses[index].status}
-            progress={clientCourses[index].progress}
-            applyDate={clientCourses[index].applyDate}
-            courseId={clientCourses[index]._id}
-            timeout={COURSE_DISABLE_DAYS}
-          />
-        ) : (
-          <CustomButton id={course._id} onClick={handleApplyCourse} variant="mediumContained">
-            {ButtonLabels.applyTheCourse}
-          </CustomButton>
-        ))}
-    </CourseActionsWrapper>
-  </CourseActionsBox>
-);
+        {!isAdmin &&
+          (clientCourses ? (
+            <ActionButton
+              label={COURSE_LABELS[clientCourses[index].status]}
+              status={clientCourses[index].status}
+              progress={clientCourses[index].progress}
+              applyDate={clientCourses[index].applyDate}
+              courseId={clientCourses[index]._id}
```

Comment:
Can you destructure this? 

```suggestion
const {_id, status, progress, applyDate} = clientCourses[index] || {}

label={COURSE_LABELS[status]}
status={status}
progress={progress}
applyDate={applyDate}
courseId={_id}
```

Reply:
@ostapdans99 this method caused eslint and ts errors such as `'Variable name _id must match one of the following formats: camelCase, PascalCase, UPPER_CASE' ` and `Type 'undefined' cannot be used as an index type.` so my solution is on the screenshoot. What do you think about this one? Should we use it or leave it without destructuring?

<img width="651" alt="image" src="https://user-images.githubusercontent.com/44285339/163862889-5916ec68-8ab2-410f-92f1-624b7c10fb95.png">

Reply:
I think, more readable variant with distructure. I dont see any problems with your solution